mappers/bluetooth_mapper/controller/controller.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"encoding/json"
+	"strings"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	"github.com/golang/glog"
+	"github.com/paypal/gatt"
+	"github.com/paypal/gatt/examples/option"
+
+	"github.com/kubeedge/kubeedge/mappers/bluetooth_mapper/action_manager"
+	"github.com/kubeedge/kubeedge/mappers/bluetooth_mapper/configuration"
+	"github.com/kubeedge/kubeedge/mappers/bluetooth_mapper/data_converter"
+	"github.com/kubeedge/kubeedge/mappers/bluetooth_mapper/helper"
+	"github.com/kubeedge/kubeedge/mappers/bluetooth_mapper/scheduler"
+	"github.com/kubeedge/kubeedge/mappers/bluetooth_mapper/watcher"
+)
+
+// constants which can be used to convey topic information
+const (
+	MapperTopicPrefix              = "$ke/mappers/bluetooth-mapper/"
+	WatcherTopicSuffix             = "/watcher/create"
+	SchedulerCreateTopicSuffix     = "/scheduler/create"
+	SchedulerDeleteTopicSuffix     = "/scheduler/delete"
+	ActionManagerCreateTopicSuffix = "/action-manager/create"
+	ActionManagerDeleteTopicSuffix = "/action-manager/delete"
+)
+
+var topicMap = make(map[string]MQTT.MessageHandler)
+
+//ControllerConfig contains the configuration used by the controller
+type ControllerConfig struct {
+	Mqtt          configuration.Mqtt                  `yaml:"mqtt"`
+	Device        configuration.Device                `yaml:"device"`
+	Watcher       watcher.Watcher              `yaml:"watcher"`
+	Scheduler     scheduler.Scheduler          `yaml:"scheduler"`
+	ActionManager action_manager.ActionManager `yaml:"action-manager"`
+	Converter     data_converter.Converter     `yaml:"data-converter"`
+}
+
+// initTopicMap initializes topics to their respective handler functions
+func (c *ControllerConfig) initTopicMap() {
+	topicMap[MapperTopicPrefix+c.Device.Id+WatcherTopicSuffix] = c.onWatcherMessageReceived
```

Comment:
can we rename onWatcherMessageReceived to handleWatchMessage. similarly for other actions ?

Reply:
done, thanks